low-level/framework/utils/conf_utils.py

```diff
@@ -148,10 +148,18 @@
 
 Conf.load(SSPL_CONF, "yaml:///etc/sspl.conf")
 global_config = Conf.get(SSPL_CONF, "SYSTEM_INFORMATION>global_config_copy_url")
-Conf.load(GLOBAL_CONF, global_config, fail_reload=False)
+if global_config:
+    Conf.load(GLOBAL_CONF, global_config, fail_reload=False)
 
-SRVNODE = Conf.get(GLOBAL_CONF, "server_node>%s>name" % MACHINE_ID)
-ENCLOSURE = Conf.get(GLOBAL_CONF, "server_node>%s>storage>enclosure_id" % MACHINE_ID)
+    SRVNODE = Conf.get(GLOBAL_CONF, "server_node>%s>name" % MACHINE_ID)
+    ENCLOSURE = Conf.get(GLOBAL_CONF, "server_node>%s>storage>enclosure_id" % MACHINE_ID)
+    STORAGE_TYPE_KEY = "storage_enclosure>%s>type" % ENCLOSURE
+    CNTRLR_PRIMARY_IP_KEY = "storage_enclosure>%s>controller>primary>ip" % ENCLOSURE
+    CNTRLR_PRIMARY_PORT_KEY = "storage_enclosure>%s>controller>primary>port" % ENCLOSURE
+    CNTRLR_SECONDARY_IP_KEY = "storage_enclosure>%s>controller>secondary>ip" % ENCLOSURE
+    CNTRLR_SECONDARY_PORT_KEY = "storage_enclosure>%s>controller>secondary>port" % ENCLOSURE
+    CNTRLR_USER_KEY = "storage_enclosure>%s>controller>user" % ENCLOSURE
+    CNTRLR_SECRET_KEY = "storage_enclosure>%s>controller>secret" % ENCLOSURE
```

Comment:
Reason of Change:
we are importing `Platform` in sspl_post_install and global_config_url is also set in `post_install : process()`.
But since imports are loaded first, conf_utils.py gets executed before we set the global_config_url.
So we need to avoid loading the file if its URL is empty in config.
In Post-install after setting the global_config_url we are loading the global config anyway, so execution will be as expected.

Reply:
Yes, so if global_config_url is empty then these variables won't be available and the class importing these variables will produce an import error. But since we are setting global_config_url in post_install, every other code importing these variables won't fail since url is available now.

Reply:
ok, so provisioner itself would fail if global_config_url not set in post-install